.jenkins/Jenkinsfile

```diff
@@ -1,21 +1,20 @@
 pipeline {
   agent any
   stages {
+    stage('Check pre-commit requirements') {
```

Comment:
Nope. I mean the original reason we moved this stage into the parallel stage and enabled fail-fast. #690 was merged by Bors, but actually failed the pre-commit-reqs requirement. When we did the post-mortem, we found that because that test was in a stage prior to the parallel build+tests stages, it was possible for a run to be restarted from the second point, essentially skipping the first stage. See [here](https://github.com/Microsoft/openenclave/pull/1012#issue-227441079).

Reply:
@andschwa Okay I see what you mean. I agree with the workaround to the problem that you suggest (by keeping the Pre-commit check in the parallel build), but underneath all this is a bug with Bors or possibly with limitations of the protocol between Bors/Jenkins.

Here are my thoughts on it: When you click the button that says "Restart this stage", it will trigger a new build (with a new build number), starting at the requested stage (instead of at the beginning of the pipeline description). The expected behavior should be that the result of that build should not interfere with the build that Bors triggered by running "bors try/r+". Bors should only look at the result of the build that it kicked off, and no others. Doing otherwise is incorrect behavior and allows people to bypass CI gating.  When Bors kicks off a build, it should ideally be able to grab the build number that it initiated, and then remember that number for when that build finishes. I'd have to look at the actual data being sent between the services to determine if this is possible with the current protocol. If it isn't, it should be :)

But this is definitely important to note, because we may eventually extend the CI system to support more stages that aren't in the giant parallel stage, and if we do that, we will expose our CI gating to this issue.

Reply:
Bors knows nothing about Jenkins or a build number; it only knows the status updated on the GitHub commit status hook that Jenkins triggers.

I think it is wrong that Jenkins (a) does not mark a build as failed when a stage is skipped and (b) updates a commit status hook with restarted builds.